app/models/audit/event/authn.rb

```diff
@@ -83,9 +83,9 @@ def attempted_action
       end
 
       def auth_stuctured_data
-        { authenticator: @authenticator_name }.tap do |sd|
+        { user: sanitized_role_id }.tap do |sd|
+          sd[:authenticator] = @authenticator_name
           sd[:service] = service_id if @service
```

Comment:
Do you want to try to rewrite the function to a more readable way so it will be one object:
```
 { 
      user: sanitized_role_id,
      authenticator: @authenticator_name,
      service: service_id if @service 
}
```